src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Item.js

```diff
@@ -1,28 +1,84 @@
 // @flow
-import Icon from '../../components/Icon';
-import type {Item as ItemType} from './types';
+import {action, observable} from 'mobx';
 import React from 'react';
+import classNames from 'classnames';
+import {observer} from 'mobx-react';
+import type {Item as ItemType} from './types';
+import Icon from '../../components/Icon';
 import itemStyles from './item.scss';
+import Dropdown from './Dropdown';
 
+@observer
 export default class Item extends React.PureComponent {
     props: ItemType;
 
     static defaultProps = {
         enabled: true,
     };
 
+    @observable hasDropdownOpen = false;
+
+    @action toggleDropdown = () => {
+        this.hasDropdownOpen = !this.hasDropdownOpen;
+    };
+
+    @action closeDropdown = () => {
+        if (this.hasSubItems) {
+            this.hasDropdownOpen = false;
+        }
+    };
+
+    componentWillMount() {
+        const {items} = this.props;
+
+        this.hasSubItems = items && items.length;
+    }
+
     handleClick = () => {
-        if (this.props.enabled && this.props.onClick) {
-            this.props.onClick();
+        const {enabled, onClick} = this.props;
+
+        if (enabled) {
+            if (onClick) {
+                onClick();
+            }
+
+            if (this.hasSubItems) {
+                this.toggleDropdown();
+            }
         }
     };
 
+    onSubItemSelected = () => {
+        this.closeDropdown();
+    };
+
     render() {
+        const { 
+            icon, 
+            items,
+            title, 
+            enabled, 
+        } = this.props;
+        const itemContainerClasses = classNames({
+            [itemStyles.container]: true,
+            [itemStyles.hasDropdownOpen]: this.hasDropdownOpen,
+        });
+
         return (
-            <button className={itemStyles.item} disabled={!this.props.enabled} onClick={this.handleClick}>
-                <Icon className={itemStyles.icon} name={this.props.icon} />
-                <span className={itemStyles.title}>{this.props.title}</span>
-            </button>
+            <div className={itemContainerClasses}>
+                <button className={itemStyles.item} disabled={!enabled} onClick={this.handleClick}>
+                    <Icon className={itemStyles.icon} name={icon} />
+                    <span className={itemStyles.title}>{title}</span>
+                    {
+                        this.hasSubItems &&
```

Comment:
I would have written these statements as described in the [react documentation](https://facebook.github.io/react/docs/conditional-rendering.html#inline-if-with-logical--operator):

```javascript
{this.hasSubItems &&
    <Icon ... />
```